### PR TITLE
Add basic github action to test build

### PR DESCRIPTION
**Pull request by utzcoz**
It adds basic GitHub Actions to test build only now.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Build
+
+on:
+  push:
+    branches: [ main ]
+
+  pull_request:
+    branches: [ main ]
```
Comment (FareesHussain) [inline .github/workflows/build.yml:8]:
```suggestion
  workflow_dispatch:
  push:
  pull_request:
    branches: 
      - main
```

Add workflow dispatch to manually run the workflow 
***merging both `pull_request` and `push` to make it short***

**Review by FareesHussain — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (utzcoz) [reply .github/workflows/build.yml]: Done. Thanks.

**Comment by utzcoz** [comment]
Hi @yrezgui , is there any update for it ? I rebased it to current `main` branch.

**Comment by utzcoz** [comment]
New running result with new spotless checking commit is here: https://github.com/utzcoz/modernstorage/runs/2785008975 .

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Build
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
```
Comment (yrezgui) [inline .github/workflows/build.yml:10]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: 8
+          distribution: 'adopt'
```
Comment (yrezgui) [inline .github/workflows/build.yml:18]:
```suggestion
```suggestion
        with:
          distribution: 'zulu'
          java-version: '11'
          java-package: jdk
```
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 8
```
Comment (yrezgui) [inline .github/workflows/build.yml:14]:
```suggestion
      - name: Set up JDK 11
```

**Review by yrezgui — CHANGES_REQUESTED** [review]
I've added some suggestions. Thanks @utzcoz !

File: .github/workflows/build.yml (hunk 2 above)
Comment (utzcoz) [reply .github/workflows/build.yml:10]: Thanks. Done.

File: .github/workflows/build.yml (hunk 3 above)
Comment (utzcoz) [reply .github/workflows/build.yml]: Thanks. Done.

File: .github/workflows/build.yml (hunk 4 above)
Comment (utzcoz) [reply .github/workflows/build.yml]: Thanks. Done.

File: .github/workflows/build.yml (hunk 2 above)
Comment (utzcoz) [reply .github/workflows/build.yml:10]: The `ubuntu-latest` maybe change when new ubuntu released, and it causes some GitHub actions failed some months ago when GitHub changed `ubuntu-latest` to `ubuntu-20.0.4`, https://github.com/actions/virtual-environments/issues/1816.

**Comment by utzcoz** [comment]
New running result is here: https://github.com/utzcoz/modernstorage/runs/2785626392.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+name: Build
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (yrezgui) [inline .github/workflows/build.yml:10]: Your comment was fair, let's set the actual value
```suggestion
    runs-on: ubuntu-20.04
```

**Review by yrezgui — CHANGES_REQUESTED** [review]

**Comment by utzcoz** [comment]
Thanks.

**Pull request merged**